galpy/util/coords.py

```diff
@@ -742,20 +742,70 @@ def cov_pmrapmdec_to_pmllpmbb(cov_pmradec,ra,dec,degree=False,epoch=2000.0):
     HISTORY:
 
        2010-04-12 - Written - Bovy (NYU)
+       2020-09-21 - Adapted for array input - Mackereth (UofT)
 
     """
     if len(cov_pmradec.shape) == 3:
-        out= numpy.zeros(cov_pmradec.shape)
-        ndata= out.shape[0]
         lb = radec_to_lb(ra,dec,degree=degree,epoch=epoch)
-        for ii in range(ndata):
-            out[ii,:,:]= cov_pmradec_to_pmllbb_single(cov_pmradec[ii,:,:],
-                                                      ra[ii],dec[ii],lb[ii,1],
-                                                      degree,epoch)
+        out = cov_pmradec_to_pmllbb_array(cov_pmradec,
+                                          ra, dec, lb[:,1],
+                                          degree, epoch)
         return out
     else:
-        l,b = radec_to_lb(ra,dec,degree=degree,epoch=epoch)
-        return cov_pmradec_to_pmllbb_single(cov_pmradec,ra,dec,b,degree,epoch)
+        lb = radec_to_lb(ra,dec,degree=degree,epoch=epoch)
+        return cov_pmradec_to_pmllbb_single(cov_pmradec,ra,dec,lb[1],degree,epoch)
```

Comment:
Could these be combined into a single case (i.e., use the same `_array` function for both a single object and multiple objects)? That would also help with the test coverage, because right now the radian input case for arrays is not tested.